### PR TITLE
Split screen

### PR DESCRIPTION
**Pull request by juljac56**
The component StudentMeeting.tsx is modified, in the folder Student.
This component is designed to display an image (downloaded from the back) of the whiteboard below the JitsiFrame. 
The Web socket Protocol is used to establish a communication between the back and the front (every 1 second, the front asks to the back if there is a new image to download and display). 
This image can be cropped, the user selects the 4 coordinates of the rectangle used to crop the image, by clicking on the image.
The coordinates are sent to the back, where the image processing is operated.

### CODE REVIEW

File: box-ui/package.json [hunk 1]
```diff
@@ -25,12 +25,14 @@
...
     "react-dom": "^17.0.2",
     "react-i18next": "^11.15.5",
+    "react-new-window": "^0.2.1",
     "react-qr-code": "^2.0.3",
     "react-qr-reader": "^2.2.1",
     "react-router-dom": "^6.2.1",
     "react-scripts": "^5.0.0",
     "react-simple-keyboard": "3.4.70",
     "reactjs-popup": "^2.0.5",
+    "sass": "^1.49.9",
```
Comment (AladdinVR) [inline box-ui/package.json:35]:
```suggestion

```

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 2]
```diff
@@ -1,7 +1,323 @@
...
+        roomName: state && state.roomName ? state.roomName : 'dty',
+        domain: state && state.domain ? state.domain : 'meeting.education',
+    });
+
+    const [selectCoord, setSelectCoord] = useState<boolean>(false);
+    const [coord, setCoord] = useState<Array<Array<number>>>([]);
+
+    //circle : svg element to display on click on the image
+    const [circles, setCircles] = useState<React.SVGProps<SVGCircleElement>[]>([]);
+    const [endCarre, setEndCarre] = useState<boolean>(false);
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:33]: you could call it allPointsSelected or something like this

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 3]
```diff
@@ -1,7 +1,323 @@
...
+    useEffect(() => {
+        getImgSize(img, false);
+    }, [img]);
+
+    useEffect(() => {
+        getImgSize(img_original, true);
+    }, [img_original]);
+
+    useEffect(() => {
+        const ws1 = new WebSocket('ws://localhost:8070/ws1');
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:75]: use .env variables to stock ws://localhost:8070/ws1

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 4]
```diff
@@ -1,7 +1,323 @@
...
+            ws1.close;
+            clearInterval(interval);
+        };
+    }, []);
+
+    // function called on click to validate the coordinates entry
+    function validerSaisie() {
+        setSelectCoord(false);
+        setEndCarre(false);
+        let c: any[] = [];
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:97]: rename c, coordinatesList for instance

File: box-ui/package.json (hunk 1 above)
Comment (djexvr) [inline box-ui/package.json:35]: Why do you need sass ?

File: box-ui/public/index.html [hunk 5]
```diff
@@ -20,7 +20,8 @@
   </head>
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root">        
-    </div>
+    <div id="root">  
+    </div>    
+
```
Comment (djexvr) [inline box-ui/public/index.html:25]: Useless empty line which creates confusing diff

File: box-ui/src/components/css/BoxHome.css [hunk 6]
```diff
@@ -8,6 +8,15 @@
     height: 100vh;
     width: 100vw;
 }
+
+.CreateMeetingComponent {
```
Comment (djexvr) [inline box-ui/src/components/css/BoxHome.css:12]: This should be in the `BoxMeeting.css` stylesheet and as a matter of fact already is as the class `.BoxMeeting`

File: box-ui/src/components/css/BoxMeeting.css [hunk 7]
```diff
@@ -29,3 +29,61 @@
...
+
+.buttonStudent {
+    color: white;
+    background-color: #141414;
+    border-radius: 8px;
+    font-weight: bold;
+    padding: 5px;
+    cursor: pointer;
+    margin: 10px;
+
```
Comment (djexvr) [inline box-ui/src/components/css/BoxMeeting.css:55]: Why the empty line

File: box-ui/src/components/css/BoxMeeting.css [hunk 8]
```diff
@@ -29,3 +29,61 @@
...
+.sectionClickSolo {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: relative;
+}
+
+.sectionClick {
+    display: flex;
+    background-color: red;
```
Comment (djexvr) [inline box-ui/src/components/css/BoxMeeting.css:79]: Is it wanted or just a debug value ?

File: box-ui/src/components/css/BoxMeeting.css [hunk 9]
```diff
@@ -29,3 +29,61 @@
 .JitsiComponent {
     flex-grow: 3;
 }
+
+.containerStudent {
```
Comment (djexvr) [inline box-ui/src/components/css/BoxMeeting.css:33]: Those rules should actually be in a separate `StudentMeeting.css` and we could perhaps create an global `Meeting.css`stylesheet for rules used in all the interfaces

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 10]
```diff
@@ -1,7 +1,323 @@
...
+            const y = coord[i][1] * rapporty;
+            c = [...c, [x, y]];
+        }
+
+        const bodyFormData = new FormData();
+
+        c.forEach((item) => {
+            bodyFormData.append('coord', item);
+        });
+        axios.post('http://localhost:8070/coord', { coord: c }).then(function () {
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:113]: same here you could use .env variable

File: box-ui/src/components/css/BoxMeeting.css [hunk 11]
```diff
@@ -29,3 +29,61 @@
...
+    flex-direction: row;
+    justify-content: center;
+    background-color: black;
+    align-items: center;
+}
+
+.buttonStudent {
+    color: white;
+    background-color: #141414;
+    border-radius: 8px;
```
Comment (djexvr) [inline box-ui/src/components/css/BoxMeeting.css:50]: For better responsiveness you should use `rem` and `em` unit as they scale with the user font size

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 12]
```diff
@@ -1,7 +1,323 @@
...
+        [],
+    );
+    const state = useLocation().state as LocationState;
+    const [information, setInformation] = useState({
+        roomName: state && state.roomName ? state.roomName : 'dty',
+        domain: state && state.domain ? state.domain : 'meeting.education',
+    });
+
+    const [selectCoord, setSelectCoord] = useState<boolean>(false);
+    const [coord, setCoord] = useState<Array<Array<number>>>([]);
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:29]:
```suggestion
    const [coord, setCoord] = useState<[number,number][]>([]);
```

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 13]
```diff
@@ -1,7 +1,323 @@
...
+                const ViewportHeight = window.innerHeight * 0.5;
+                const width = newImg.width;
+                const pot_height = (height / width) * ViewportWidth;
+                if (pot_height < ViewportHeight) {
+                    height = pot_height;
+                } else {
+                    height = ViewportHeight;
+                }
+                setHeight(height);
+            }
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:192]:
```suggestion
            let height = newImg.height;
            const width = newImg.width;
            if (original) {
                const ratio_img = ((width * 50) / height).toString() + 'vh';
                setRatioOriginal(ratio_img);
                setWidthOriginal(width);
                setHeightOriginal(height);
            } else {
                // dimensions of the original image, not of the image displayed on screen
                const ViewportWidth = window.innerWidth * 0.9;
                const ViewportHeight = window.innerHeight * 0.5;
                const pot_height = (height / width) * ViewportWidth;
                if (pot_height < ViewportHeight) {
                    height = pot_height;
                } else {
                    height = ViewportHeight;
                }
                setHeight(height);
            }
```

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 14]
```diff
@@ -1,7 +1,323 @@
...
+            setImg(image_Slice.src);
+            setLoading(false);
+            setTextBtn('Recadrer');
+        });
+    }
+
+    //download the original image from the back
+    function requestOriginalImage() {
+        setLoading(true);
+        axios.get('http://localhost:8070/original_photo').then((resp) => {
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:213]: same .env comment

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 15]
```diff
@@ -1,7 +1,323 @@
...
+    // function called on click to validate the coordinates entry
+    function validerSaisie() {
+        setSelectCoord(false);
+        setEndCarre(false);
+        let c: any[] = [];
+
+        // operation to send the correct coordinates to the back
+        const rapportx = width_img_original / widthAff;
+        const rapporty = height_img_original / heightAff;
+        for (let i = 0; i < 4; i++) {
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:102]:
```suggestion
        c = coord.map((point) => [point[0] * rapportx, point[1] * rapporty])
```
You could also use a clearer name for `c`

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 16]
```diff
@@ -1,7 +1,323 @@
...
+            const x = coord[i][0] * rapportx;
+            const y = coord[i][1] * rapporty;
+            c = [...c, [x, y]];
+        }
+
+        const bodyFormData = new FormData();
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:108]: I don't get the use for this variable

File: box-ui/src/components/ts/Student/StudentMeeting.tsx (hunk 10 above)
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:113]: You should use a env variable for the url

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 17]
```diff
@@ -1,7 +1,323 @@
...
+        });
+        axios.post('http://localhost:8070/coord', { coord: c }).then(function () {
+            setCoord([]);
+            setCircles([]);
+        });
+    }
+
+    // get relative coodinates of the click of the user
+    const getClickCoords = (event: { target: any; clientX: number; clientY: number }) => {
+        const e = event.target;
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:121]: You don't need this variable you could define `dim` directly

File: box-ui/src/components/ts/Student/StudentMeeting.tsx (hunk 2 above)
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:33]: You can just check the length of `coord` to get rid of this state

**Review by djexvr — CHANGES_REQUESTED** [review]

**Pull request merged**